mne/preprocessing/ica.py

```diff
@@ -1093,9 +1086,6 @@ def find_bads_eog(self, inst, ch_name=None, threshold=3.0,
         # get targets befor equalizing
         targets = [self._check_target(k, inst, start, stop) for k in eog_chs]
 
-        if inst.ch_names != self.ch_names:
-            inst = inst.copy().pick_channels(self.ch_names)
```

Comment:
The tests already cover the case where the ica is fitted with picks. I also added tests where raw has channels dropped after fitting and raises an error. I can't imagine of any other use cases.